src/main/java/org/trigon/hardware/rev/spark/io/SimulationSparkIO.java

```diff
@@ -63,51 +74,45 @@ public void setInverted(boolean inverted) {
     }
 
     @Override
-    public void enableVoltageCompensation(double voltage) {
-        motor.enableVoltageCompensation(voltage);
+    public void setBrake(boolean brake) {
+        SparkMaxConfig configuration = new SparkMaxConfig();
+        configuration.idleMode(brake ? SparkMaxConfig.IdleMode.kBrake : SparkMaxConfig.IdleMode.kCoast);
+        motor.configure(configuration, SparkBase.ResetMode.kNoResetSafeParameters, SparkBase.PersistMode.kNoPersistParameters);
     }
 
     @Override
-    public void setClosedLoopRampRate(double rampRate) {
-        motor.setClosedLoopRampRate(rampRate);
-    }
+    public void updateSimulation() {
+        if (physicsSimulation == null)
+            return;
 
-    @Override
-    public void setSmartCurrentLimit(int limit) {
-        motor.setSmartCurrentLimit(limit);
-    }
+        physicsSimulation.setInputVoltage(motorSimulation.getBusVoltage());
+        physicsSimulation.updateMotor();
 
-    @Override
-    public void setOpenLoopRampRate(double rampRate) {
-        motor.setOpenLoopRampRate(rampRate);
+        motorSimulation.iterate(physicsSimulation.getRotorVelocityRotationsPerSecond(), RobotHardwareStats.SUPPLY_VOLTAGE, RobotHardwareStats.getPeriodicTimeSeconds());
+        if (isUsingAbsoluteEncoder()) {
+            absoluteEncoderSimulation.iterate(physicsSimulation.getSystemVelocityRotationsPerSecond(), RobotHardwareStats.getPeriodicTimeSeconds());
+            return;
+        }
+        relativeEncoderSimulation.iterate(physicsSimulation.getRotorVelocityRotationsPerSecond(), RobotHardwareStats.getPeriodicTimeSeconds());
     }
 
     @Override
-    public void setPID(double p, double i, double d) {
-        pidController.setP(p);
-        pidController.setI(i);
-        pidController.setD(d);
+    public void configure(SparkBaseConfig configuration, SparkBase.ResetMode resetMode, SparkBase.PersistMode persistMode) {
+        motor.configure(configuration, resetMode, persistMode);
     }
 
     @Override
-    public void setConversionsFactor(double conversionsFactor) {
-        encoder.setConversionsFactor(conversionsFactor);
+    public void setPhysicsSimulation(MotorPhysicsSimulation physicsSimulation, boolean isUsingAbsoluteEncoder) {
+        motorSimulation = new SparkSim(motor, physicsSimulation.getGearbox());
+        this.physicsSimulation = physicsSimulation;
+        if (isUsingAbsoluteEncoder) {
+            absoluteEncoderSimulation = new SparkAbsoluteEncoderSim(motor);
+            return;
+        }
+        relativeEncoderSimulation = new SparkRelativeEncoderSim(motor);
```

Comment:
I see a world where this function is called more than once (different sim for having load and without load)
make sure you can handle that